# Exploring the trends in the ATP Tennis Players Performance: 2017-19

# The Top 500 Players Game Results from 2017-2019 in the Association of Tennis Professionals (ATP) Tour.

The data frame that our group chose provides details about the top 500 matches from 2017- 2019 in the Association of Tennis Professionals (ATP) Tour.  The ATP Tour is organized by the Association of Tennis Professionals, which is the only men's professional tennis tour in the world.  The Excel sheet includes various information, ranging from the winners’ and losers’ names, heights, countries, scores, etc. 

The data frame provides us with lots of variables such as winner player's id, height, age, country and a lot of other information. From this we will try to answer the following question:

Can we classify the country of origin (IOC Code) of the winning tennis players based on various predictors such as player ID, surface type, player name, height and age?

So we want to build a classification model that predits the country from which a winning tennis player originates using the features such as player ID, surface type of the  match, player name, player height and player age. The goal from the activity is to understand if there is a pattern in these fetures that can help us classify the country of origin of the winning player?

In [31]:
library(tidyverse)


In [61]:
library(readr)
library(dplyr)

# Read the dataset from the web into R
tennis_players <- read_csv("data/tennis_players.csv")

# Select specific columns and remove unnamed column if present
tennis_players <- tennis_players |>
                 select(surface:winner_age)|>
select(-draw_size:-winner_entry)
                   

# Print the resulting data frame
tennis_players

New names:
• `` -> `...1`
Rows: 6866 Columns: 50
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (16): tourney_id, tourney_name, surface, tourney_level, winner_seed, win...
dbl (34): ...1, draw_size, tourney_date, match_num, winner_id, winner_ht, wi...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


surface winner_name        winner_hand winner_ht winner_ioc winner_age
1    Hard    Kei Nishikori      R           178       JPN        29.00479  
2    Hard    Daniil Medvedev    R            NA       RUS        22.88569  
3    Hard    Kei Nishikori      R           178       JPN        29.00479  
⋮    ⋮       ⋮                  ⋮           ⋮         ⋮          ⋮         
6864 Hard    Jo Wilfried Tsonga R           188       FRA        32.59411  
6865 Hard    David Goffin       R           163       BEL        26.95414  
6866 Hard    Lucas Pouille      R           185       FRA        23.73990

From the above code we read a CSV file containing tennis player data inot R using 'read_csv()' function from the 'readr' package. SO when the data frame got printed we are able to see 6866 rows and 50 columns. From that we removed the unwanted varibles related to loser's data, and then we will try to predict the country of the winner, based on differnt predictors lised above. 

### As now we have the data we will try to tidy it.


We can tidy any data using following ways:
- Checking the missing values using 'is.na()' function.
- Checking if the dataset is in wide or long format and reshape it if needed using functions like 'pivot_longer()' or 'pivot_wider()'.
- We should also check if all the varibles are in correct type, like 'dbl(double)', 'fct(factor)' or 'chr(character)'

In [64]:
# we will start with renaming the column names:
tennis_players_short_data <- tennis_players|>
rename(winner_name = winner_name,
        winner_hand= winner_hand,
        winner_height = winner_ht,
        winner_country = winner_ioc,
        winner_age= winner_age)
tennis_players_short_data

surface winner_name        winner_hand winner_height winner_country
1    Hard    Kei Nishikori      R           178           JPN           
2    Hard    Daniil Medvedev    R            NA           RUS           
3    Hard    Kei Nishikori      R           178           JPN           
⋮    ⋮       ⋮                  ⋮           ⋮             ⋮             
6864 Hard    Jo Wilfried Tsonga R           188           FRA           
6865 Hard    David Goffin       R           163           BEL           
6866 Hard    Lucas Pouille      R           185           FRA           
     winner_age
1    29.00479  
2    22.88569  
3    29.00479  
⋮    ⋮         
6864 32.59411  
6865 26.95414  
6866 23.73990